lib/client/rpcClient.ts

```diff
@@ -0,0 +1,55 @@
+import { Program, Provider } from "@coral-xyz/anchor";
+import { AutocratProgram, Dao, DaoState, ProgramVersion } from "../types";
+import { FutarchyClient } from "./client";
+import { enrichTokenMetadata } from "../tokens";
+
+export class FutarchyRPClient implements FutarchyClient {
+  private programVersion: ProgramVersion;
+  private autocratProgram: Program<AutocratProgram>;
+  private rpcProvider: Provider;
+  private constructor(programVersion: ProgramVersion, rpcProvider: Provider) {
+    this.programVersion = programVersion;
+    this.rpcProvider = rpcProvider;
+    this.autocratProgram = new Program<AutocratProgram>(
+      programVersion.idl as AutocratProgram,
+      programVersion.programId,
+      this.rpcProvider
+    );
+  }
+  static make(programVersion: ProgramVersion, rpcProvider: Provider) {
+    return new FutarchyRPClient(programVersion, rpcProvider);
+  }
+  async fetchDao(daoAddress: string): Promise<Dao | undefined> {
+    const daoAndTreasury = await this.fetchDaoState(daoAddress);
+    if (daoAndTreasury) {
+      const daoState = daoAndTreasury;
+      const baseMint = ["V0.2", "V0.3"].includes(this.programVersion.label)
+        ? daoState.tokenMint
+        : daoState.metaMint;
+      const quoteMint = daoState.usdcMint;
+      if (baseMint) {
+        const baseToken = await enrichTokenMetadata(baseMint, this.rpcProvider);
+        const quoteToken = await enrichTokenMetadata(
+          quoteMint,
+          this.rpcProvider
+        );
+        return {
+          daoState,
+          baseToken,
+          quoteToken,
+        };
+      }
+    }
+  }
+
+  private async fetchDaoState(
+    daoAddress: string
+  ): Promise<DaoState | undefined> {
+    const allDaos = await this.autocratProgram.account.dao.all();
```

Comment:
I think this should be a fetchAllDaos as we'll need it for legacy support, and likely since it fetches them all we don't need to call this for each DAO we'd just run fetchDao on each of the ones we fetched.

Reply:
Okay added a method for fetch all DAOs, and then individual dao just calls fetch(daoAddress).

Lmk what you think on this.